src/main/java/io/confluent/connect/jdbc/TimestampIncrementingTableQuerier.java

```
@@ -59,25 +59,31 @@
   private String timestampColumn;
   private Long timestampOffset;
   private String incrementingColumn;
+  private boolean incrementingColumnUsePrimaryKey;
   private Long incrementingOffset = null;
   private long timestampDelay;
 
-  public TimestampIncrementingTableQuerier(QueryMode mode, String name, String topicPrefix,
+  public TimestampIncrementingTableQuerier(QueryMode mode, String name, String topicPrefix, String keyColumnName,
                                            String timestampColumn, Long timestampOffset,
-                                           String incrementingColumn, Long incrementingOffset, Long timestampDelay) {
-    super(mode, name, topicPrefix);
+                                           String incrementingColumn, boolean incrementingColumnUsePrimaryKey, Long incrementingOffset, Long timestampDelay) {
+    super(mode, name, topicPrefix, keyColumnName);
     this.timestampColumn = timestampColumn;
     this.timestampOffset = timestampOffset;
-    this.incrementingColumn = incrementingColumn;
     this.incrementingOffset = incrementingOffset;
     this.timestampDelay = timestampDelay;
+    this.incrementingColumn = incrementingColumn;
+    this.incrementingColumnUsePrimaryKey = incrementingColumnUsePrimaryKey;
   }
 
   @Override
   protected void createPreparedStatement(Connection db) throws SQLException {
-    // Default when unspecified uses an autoincrementing column
+    // Default when unspecified uses an autoincrementing column or primary key if primary key assumed to autoincrement.
     if (incrementingColumn != null && incrementingColumn.isEmpty()) {
-      incrementingColumn = JdbcUtils.getAutoincrementColumn(db, name);
+      if (incrementingColumn.isEmpty() && incrementingColumnUsePrimaryKey && keyColumn != null) {
```

This duplicates the `incrementingColumn.isEmpty()` check that was already done for the `if` right above this.

Also, I think this is at least one location we could throw the exception if `incrementingColumnUsePrimaryKey` is true but `keyColumn` is false, right? Although maybe that's better done as a check in the constructor so we trigger the error as early as possible?